src/utils/futures.rs

```diff
@@ -285,6 +285,30 @@ impl Drop for CallOnDrop {
     }
 }
 
+/// Error returned by [`timeout_compat`].
+#[derive(Debug, PartialEq, thiserror::Error)]
+#[error("deadline has elapsed")]
+pub struct Elapsed(());
```

Comment:
huh, i always wrote stuff like this as `struct Elapsed { }`.  guess it doesn't matter

Reply:
So did I, but turns out you can construct such a type.

Reply:
… Now that you mention it, why not `struct Elapsed;` without any members. I think that makes the intention clearer, or do you actually need a member?

Reply:
I copied tokio's [`Elapsed`](https://docs.rs/tokio/1.0.1/tokio/time/error/struct.Elapsed.html), that's why. But the reason is always the same: Prevent that the type is constructed by a user.